service/interpreter/workflowImpl.go

```diff
@@ -808,6 +806,38 @@ func invokeStateExecute(
 	return &decision, service.CompletedStateExecutionStatus, nil
 }
 
+func failOnSignalWithStart(state iwfidl.StateMovement, provider WorkflowProvider, err error) (*iwfidl.StateDecision, service.StateExecutionStatus, error) {
+	if shouldProceedOnExecuteApiError(state) {
+		return nil, service.ExecuteApiFailedAndProceed, nil
+	}
+	return nil, service.FailureStateExecutionStatus, convertStateApiActivityError(provider, err)
+}
+
+func signalWithStartWorkflowId(state iwfidl.StateMovement, parentId string, executionContext iwfidl.Context) string {
+	if state.WaitForKey != nil {
+		return service.IwfSystemConstPrefix + parentId + "_" + state.StateId + "_" + *state.WaitForKey
+	} else {
+		return service.IwfSystemConstPrefix + parentId + "_" + *executionContext.StateExecutionId
+	}
+}
+
+func signalWithStart(unifiedClient uclient.UnifiedClient, workflowId string) error {
+	err := unifiedClient.SignalWithStartWaitForStateCompletionWorkflow(
+		context.Background(),
+		uclient.StartWorkflowOptions{
+			ID:                       workflowId,
+			TaskQueue:                env.GetTaskQueue(),
+			WorkflowExecutionTimeout: 60 * time.Second, // timeout doesn't matter here as it will complete immediate with the signal
+		},
+		iwfidl.StateCompletionOutput{})
+	if err != nil && !unifiedClient.IsWorkflowAlreadyStartedError(err) {
+		// WorkflowAlreadyStartedError is returned when the started workflow is closed and the signal is not sent
+		// panic will let the workflow task will retry until the signal is sent
+		panic(fmt.Errorf("failed to signal on completion %w", err))
+	}
+	return err
```

Comment:
I think this func doesn't need to return anything(error).

It's a bit tricky here that we need to expand on later about "non determinism and how workflow task is executed". 

For here, we intentionally swallow the  error on "alreadyStarted", and panic on any other errors. By panicing, it will just retry automatically. 

We don't want to proceed to other states for any other errors.

Reply:
Fixed the issue